tests/utilities.py

```diff
@@ -226,6 +226,139 @@ def get_user_next_id(remote_id):
     return next_id
 
 
+def wait_for_prpsl_rjctn_in_db(object_id, max_attempts=10, delay=0.5):
+    """Polls rethinkdb for the requested proposal until it has been rejected.
+    Useful when commiting a delete transaction in sawtooth and waiting for the
+    related proposals to be rejected for dependent chained transactions.
+
+    Args:
+        object_id
+            str:    Object_id of the proposal to wait for.
+        max_attempts:
+            int:    The number of times to attempt to find the given role before
+                    giving up and returning False.
+                    Default value: 10
+        delay:
+            float:  The number of seconds to wait between query attempts.
+                    Default value: 0.5
+    Returns:
+        resource_removed:
+            bool:
+                True:   If the role is successfully found within the given
+                        number of attempts.
+            bool:
+                False:  If the role is not found after the given number of
+                        attempts.
+    """
+    is_proposal_closed = False
+    count = 0
+    with connect_to_db() as conn:
+        while not is_proposal_closed and count < max_attempts:
+            resource = (
+                r.table("proposals")
+                .filter({"object_id": object_id, "status": "REJECTED"})
+                .coerce_to("array")
+                .run(conn)
+            )
+            LOGGER.info(resource)
```

Comment:
Logger left here on accident? Did we wanted to keep logging here to help us see the outcome of our tests?